### PR TITLE
Added layout structure and changed routing method

### PR DESCRIPTION
**Pull request by SreenathES**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,13 +1,58 @@
...
+      {
+        path: 'search/:q',
+        element: <CourseSearch />
+      }
+    ]
+  },
+  {
+    path: 'student',
+    element: <StudentLayout />,
+    children: [
```
Comment (mishelshaji) [inline client/src/App.js]: Remove the children array if empty.

File: client/src/App.js [hunk 2]
```diff
@@ -1,13 +1,58 @@
...
+      {
+
+      }
+    ]
+  },
+  {
+    path: 'instructor',
+    element: <InstructorLayout />,
+    children: [
```
Comment (mishelshaji) [inline client/src/App.js]: Remove the children array if empty.

File: client/src/App.js [hunk 3]
```diff
@@ -1,13 +1,58 @@
...
+      {
+
+      }
+    ]
+  },
+  {
+    path: 'admin',
+    element: <AdminLayout />,
+    children: [
```
Comment (mishelshaji) [inline client/src/App.js]: Remove the children array if empty.

File: client/src/pages/UserLayout.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { Outlet } from 'react-router-dom';
+import NavigationBar from "../components/NavigationBar";
+import Footer from '../components/Footer';
+
+function UserLayout() {
+    return (
+        <>
+            <NavigationBar />
```
Comment (mishelshaji) [inline client/src/pages/UserLayout.js]: Format the code.

**Review by mishelshaji — CHANGES_REQUESTED** [review]

File: client/src/App.js (hunk 1 above)
Comment (SreenathES) [reply client/src/App.js]: removed children array

File: client/src/App.js (hunk 3 above)
Comment (SreenathES) [reply client/src/App.js]: removed children array

File: client/src/App.js (hunk 2 above)
Comment (SreenathES) [reply client/src/App.js]: removed children array

File: client/src/pages/UserLayout.js (hunk 4 above)
Comment (SreenathES) [reply client/src/pages/UserLayout.js]: code formatted

**Review by mishelshaji — APPROVED** [review]

**Pull request merged**